### PR TITLE
more docs

### PR DESCRIPTION
**Pull request by KristianKjerstad**
## Why is this pull request needed?

add more docs about using Git and setting up debugger

## What does this pull request change?

docs
## Issues related to this change:

### CODE REVIEW

File: docs/getting-started/inital-setup.md
Comment (soofstad) [inline docs/getting-started/inital-setup.md:106]:
```suggestion
   3) python interpreter: the newly created docker compose interpreter (step 4)
```

File: docs/getting-started/prerequisites.md
Comment (soofstad) [inline docs/getting-started/prerequisites.md:14]: Could put "python" and "node" under a separate "If you want to run locally".
Could then also add "poetry" and "yarn"

File: docs/how-to/using-git.md
Comment (soofstad) [inline docs/how-to/using-git.md:59]:
```suggestion
* All tests passes
```

File: docs/how-to/using-git.md
Comment (soofstad) [inline docs/how-to/using-git.md:60]:
```suggestion
```

File: docs/how-to/using-git.md
Comment (soofstad) [inline docs/how-to/using-git.md:66]:
```suggestion
* Readability: Would another developer be able to easily understand and work with the code? Are there none-idiomatic patterns?
```

**Review by soofstad — CHANGES_REQUESTED** [review]

File: docs/how-to/using-git.md
Comment (mariuswinger) [inline docs/how-to/using-git.md:45]: What does <optional scope> refer to in this context?

File: docs/how-to/using-git.md
Comment (mariuswinger) [inline docs/how-to/using-git.md:47]: typo: necessary. Rephrase: "When necessary" -> "If necessary"; "until 72" -> "up to 72 on each line"

File: docs/how-to/using-git.md
Comment (mariuswinger) [inline docs/how-to/using-git.md:58]: The commit messages above are not described as "conventional" meaning that the word conventional in this list loses meaning unless you already know what it means. I suggest adding the word conventional in the paragraph above: "Commit messages should follow the conventional commit message specification". We could also consider linking to the documentation of conventional commit there-

File: docs/how-to/using-git.md
Comment (mariuswinger) [inline docs/how-to/using-git.md:69]:
```suggestion
* Comments: Are comments understandable and useful?
```

File: docs/how-to/using-git.md
Comment (mariuswinger) [inline docs/how-to/using-git.md:13]:
```suggestion
Whenever a developer makes changes to the code, a new Pull Request (PR) should be created. The new PR should branch out from the main branch.
```

File: docs/how-to/using-git.md
Comment (mariuswinger) [reply docs/how-to/using-git.md:45]: Also, it is not evident if type refers to the same types as above, or if this type parameter includes more or less types than above.

File: docs/how-to/using-git.md
Comment (olavis) [inline docs/how-to/using-git.md:38]: Do we need all these types? Only `fix` and `feat` are mandated by the conventional commits specification. I've only found half of these types useful. If the type isn't a fix or feat I always find the prefix the least useful part of the commit message/branch name. Am I the only one?

File: docs/how-to/using-git.md
Comment (olavis) [inline docs/how-to/using-git.md:41]: Maybe we can add some examples or do's/dont's? For instance messages like: `fixed comments from review` may seem informative within the scope of the PR, but are not really useful when you need to debug or go through the git history later.

File: docs/how-to/using-git.md
Comment (olavis) [inline docs/how-to/using-git.md:57]: This section and `What to check in code reviews` should be merged. The team share equal responsibility for the code, and we should review our own code before asking team members.

File: docs/how-to/using-git.md
Comment (olavis) [inline docs/how-to/using-git.md:68]: + following correct naming conventions, casing, aim for names to be _exact_ - makes it easier to avoid violation of the the single responsibility principle.

File: docs/how-to/using-git.md
Comment (olavis) [inline docs/how-to/using-git.md:63]: Maybe we can list the SOLID principles, to have something thats more easily measured than `is the code well-designed?`

File: docs/how-to/using-git.md
Comment (olavis) [inline docs/how-to/using-git.md:13]: developer

Comment (olavis) [reply docs/how-to/using-git.md:13]: Also, not sure if I agree that a new PR should always branch out from main. It can be very useful to branch out separate issues from a topic branch and create a PR into the given branch instead of just making a single (huge) PR that contains multiple features/fixes.

File: docs/getting-started/inital-setup.md
Comment (olavis) [inline docs/getting-started/inital-setup.md:102]: Configuration

File: docs/how-to/using-git.md
Comment (olavis) [inline docs/how-to/using-git.md:19]: `When a PR is approved` What if the PR isn't approved? Or do we take it for granted? Ideally we should have some guidelines for when to close a PR, if it's growing too complex or has gotten outdated due to staying open for too long. A PR can affect the effectiveness of the team badly if it remains open.

**Comment by KristianKjerstad** [comment]
todo
should add info about env vars to docs (see for example https://github.com/equinor/QuatoOps/blob/master/docs/getting_started.md )

**Comment by KristianKjerstad** [comment]
note: some github repos add a description of the project file structure, which can be useful but also difficult to maintain

https://github.com/microsoft/MLOpsPython/blob/master/docs/code_description.md

File: docs/getting-started/prerequisites.md
Comment (soofstad) [inline docs/getting-started/prerequisites.md:14]:
```suggestion
If you want to run tests, pre-commit etc. outside of docker, you also need:
```

**Review by soofstad — APPROVED** [review]

**Review by sebastianvitterso — APPROVED** [review]

**Comment by KristianKjerstad** [comment]
outdated, see #78

**Pull request closed without merging**
